src/permutation.jl

```diff
@@ -33,10 +33,15 @@ Perform a permutation test (a.k.a. randomization test) of the null hypothesis
 that `f(x)` is equal to `f(y)`.  `n` of the `factorial(length(x)+length(y))`
 permutations are sampled at random.
 """
-function ApproximatePermutationTest(x::AbstractVector{R}, y::AbstractVector{S},
-                                    f::Function, n::Int) where {R<:Real,S<:Real}
+function ApproximatePermutationTest(
+    x::AbstractVector{R},
+    y::AbstractVector{S},
+    f::Function,
+    n::Int;
+    rng::AbstractRNG=Random.default_rng()
+) where {R<:Real,S<:Real}
```

Comment:
Can you keep the same style as in the rest of the file? I'm not sure how to indent the last line though.
```suggestion
function ApproximatePermutationTest(x::AbstractVector{R},
                                    y::AbstractVector{S},
                                    f::Function,
                                    n::Int;
                                    rng::AbstractRNG=Random.default_rng()) where
                                    {R<:Real,S<:Real}
```

Reply:
I've made the indentation in this file consistent. The old way of doing it wastes a lot of horizontal space.

Reply:
Wait, none of the other files in the project (nor in JuliaStats packages AFAIK) use this style, so better stick to the previous one. If we decide to change styles we should discuss this first.

Reply:
The new style was discussed and adopted six years ago in https://github.com/JuliaStats/GLM.jl/pull/126 :trollface: I propose that we gradually move towards that style as we touch the files for other reasons. It really is a nicer formatting for functions with many arguments.

Reply:
Styles are the worst thing to discuss so I'd rather stick to what we have now, unless we apply an autoformatter to all our codebase so that we no longer have to deal with these issues.